### PR TITLE
common component image

### PR DESCRIPTION
**Pull request by mFlaifel**
image component that accept many props:
source, width,height, onClick, alt, style.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -4,9 +4,12 @@
   "private": true,
   "proxy": "http://localhost:5000",
   "dependencies": {
+    "@material-ui/core": "^4.9.9",
+    "@material-ui/icons": "^4.9.1",
```
Comment (Asmaathabet) [inline client/package.json:8]: if you made pull to the master , you will found that they are already installed ^_^

File: client/src/components/common/Image/index.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  animationDuration: PropTypes.number,
+  onClick: PropTypes.func,
+  alt: PropTypes.string,
+  style: PropTypes.string,
+};
+CommonImage.defaultProps = {
+  animationDuration: 0,
+  width: '100%',
+  height: 'auto',
+  onClick: null,
```
Comment (Asmaathabet) [inline client/src/components/common/Image/index.js:37]:
```suggestion
  onClick: () => {},
```

File: client/src/components/common/Image/index.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  alt: 'application custom image',
+  style: null,
```
Comment (Asmaathabet) [inline client/src/components/common/Image/index.js:39]:
```suggestion
  style: ' ',
```

File: client/src/components/common/Image/index.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import Image from 'material-ui-image';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const CommonImage = ({
+  src,
+  width,
+  height,
+  animationDuration,
```
Comment (Asmaathabet) [inline client/src/components/common/Image/index.js:9]: there is no need for this cause we will not use animation

File: client/src/components/common/Image/index.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  onClick,
+  alt,
+  style,
+}) => (
+  <Image
+    imageStyle={{ width, height }}
+    animationDuration={animationDuration}
+    onClick={onClick}
+    alt={alt}
+    style={style}
```
Comment (Asmaathabet) [inline client/src/components/common/Image/index.js:19]: There is ready props called image style to override image style , so I think its good to use it .
https://mui.wertarbyte.com/#image 
```suggestion
    imageStyle={style}
```

File: client/src/components/common/Image/index.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+);
+
+CommonImage.propTypes = {
+  src: PropTypes.string.isRequired,
+  width: PropTypes.string,
+  height: PropTypes.string,
+  animationDuration: PropTypes.number,
+  onClick: PropTypes.func,
+  alt: PropTypes.string,
+  style: PropTypes.string,
```
Comment (Asmaathabet) [inline client/src/components/common/Image/index.js:31]: please arrange proptypes by the same way in all places & put the `src` at the first in all cases.

File: client/src/components/common/Image/index.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+};
+CommonImage.defaultProps = {
+  animationDuration: 0,
+  width: '100%',
+  height: 'auto',
+  onClick: null,
+  alt: 'application custom image',
+  style: null,
+};
```
Comment (Asmaathabet) [inline client/src/components/common/Image/index.js:40]: Also, you can create a file & add a custom style for it, for example to use for one images that we have , put the style that made it as we want to look . so this style will be ready to use later as it is .

File: client/src/components/common/Image/index.js (hunk 5 above)
Comment (mFlaifel) [reply client/src/components/common/Image/index.js:19]: there is two ready props "style and imageStyle",
* style is used to write any style you want.
* imageStyle to specify just width and height.

[Material UI Image](https://github.com/TeamWertarbyte/material-ui-image/blob/v3.1.0/README.md#material-ui-image-properties)

File: client/src/components/common/Image/index.js (hunk 3 above)
Comment (mFlaifel) [reply client/src/components/common/Image/index.js:29]: when i try to change style to empty string the app crashes

File: client/src/components/common/Image/index.js (hunk 5 above)
Comment (Asmaathabet) [reply client/src/components/common/Image/index.js:11]: ok, got it. thank you

**Pull request merged**